browser-sign-in/OktaBrowserSignIn/WelcomeViewController.swift

```diff
@@ -15,70 +15,88 @@
-            // We'll allow the switch to be shown
+        if let configForUITests = configForUITests {
+            self.auth = WebAuthentication(
+                issuer: URL(string: configForUITests["issuer"]!)!,
+                clientId: configForUITests["clientId"]!,
+                scopes: configForUITests["scopes"]!,
+                redirectUri: URL(string: configForUITests["redirectUri"]!)!,
+                logoutRedirectUri: URL(string: configForUITests["logoutRedirectUri"] ?? ""),
+                additionalParameters: nil)
         } else {
-            ephemeralSwitch.superview?.isHidden = true
+            self.auth = WebAuthentication.shared
         }
         
-        do {
-            if let configForUITests = self.configForUITests {
-                oktaOidc = try OktaOidc(configuration: OktaOidcConfig(with: configForUITests))
-            } else {
-                oktaOidc = try OktaOidc()
-            }
-        } catch let error {
-            DispatchQueue.main.async {
-                let alert = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
-                alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
-                self.present(alert, animated: true, completion: nil)
-            }
-            return
+        if Credential.default != nil {
+            self.performSegue(withIdentifier: "show-details", sender: self)
         }
         
-        if  let oktaOidc = oktaOidc,
-            let _ = OktaOidcStateManager.readFromSecureStorage(for: oktaOidc.configuration)?.accessToken {
-            self.stateManager = OktaOidcStateManager.readFromSecureStorage(for: oktaOidc.configuration)
-            performSegue(withIdentifier: "show-details", sender: self)
+        if let clientId = auth?.signInFlow.client.configuration.clientId {
+            self.clientIdLabel.text = "clientId: \(clientId)"
+        } else {
+            self.clientIdLabel.text = "Not configured"
+            self.signInButton.isEnabled = false
         }
     }
-
+    
+    func navigateToDetailsPage() {
+        guard Credential.default != nil else { return }
+        self.performSegue(withIdentifier: "show-details", sender: self)
+    }
+    
+    @IBAction func ephemeralSwitchChanged(_ sender: Any) {
+        guard let sender = sender as? UISwitch else { return }
+        self.auth?.ephemeralSession = sender.isOn
+    }
+    
     @IBAction private func signInTapped() {
-        if #available(iOS 13.0, *) {
-            oktaOidc?.configuration.noSSO = ephemeralSwitch.isOn
-        }
-        
-        oktaOidc?.signInWithBrowser(from: self, callback: { [weak self] stateManager, error in
-            if let error = error {
-                let alert = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
+        let window = viewIfLoaded?.window
+        auth?.signIn(from: window) { result in
```

Comment:
No need to use `viewIfLoaded` since this method will only be invoked when the view _is_ loaded.

```suggestion
        auth?.signIn(from: view.window) { result in
```